rfc/src/rfcs/0007-global-conditions.md

```diff
@@ -0,0 +1,105 @@
+In particular, the addition of a new option to fail verification if there are uncoverable (i.e., unsatisfiable or unreachable) `cover` properties (requested in [#2299](https://github.com/model-checking/kani/issues/2299)) is posing new challenges to our current architecture and UI.
+
+This concept isn't made explicit in Kani, but exists in some ways.
+For example, the `kani::should_panic` attribute is a global condition because it can be described in terms of other properties (checks).
+The request in [#2299](https://github.com/model-checking/kani/issues/2299) is essentially another global conditions, and we may expect more to be requested in the future.
+
+In this RFC, we propose a new section in Kani's output focused on reporting global conditions.
+The goal is for users to receive useful information about hyperproperties without it becoming overwhelming.
+This will help users to understand better options that are enabled through global conditions and ease the addition of such options to Kani.
+
+## User Experience
+
+**The output will refer to properties that depend on other properties as "global conditions"**, which is a simpler term.
+The options to enable different global conditions will depend on a case-by-case basis.
+
+The main UI change in this proposal is a new `GLOBAL CONDITIONS` section that **won't be printed if no global conditions have been enabled**.
+This section will only appear in Kani's default output after the `RESULTS` section (used for individual checks) and have the format:
+
+```
+GLOBAL CONDITIONS:
+ - `<name>`: <status> (<reason>)
+ - `<name>`: <status> (<reason>)
+ [...]
+```
+
+where:
+  - `<name>` is the name given to the global condition.
+  - `<status>` is the status determined for the global condition.
+  - `<reason>` is an explanation that depends on the status of the global condition.
+
+For example, let's assume we implement the option requested in [#2299](https://github.com/model-checking/kani/issues/2299).
+A concrete example of this output would be:
+
+```
+GLOBAL CONDITIONS:
+ - `fail_uncoverable`: SUCCESS (all cover statements were satisfied as expected)
+```
+
+A `FAILED` status in any enabled global condition will cause verification to fail.
+In that case, the overall verification result will point out that one or more global conditions failed, as in:
+
+```
+VERIFICATION:- FAILURE (one or more global conditions failed)
+```
+
+This last UI change will also be implemented for the terse output.
+Finally, checks that cause an enabled global condition to fail will be reported using the same interface we use for failed checks[^failed-checks].
+
+**Global conditions which aren't enabled won't appear in the `GLOBAL CONDITIONS` section**.
+Their status will be computed regardless[^status-computation], and we may consider showing this status when the `--verbose` option is passed.
+
+The documentation of global conditions will depend on how they're enabled, which depends on a case-by-case basis.
```

Comment:
How will users specify and enable global conditions? Can you give an example here?

Reply:
Added a clarification regarding this. As mentioned, global conditions won't force a single mechanism to be enabled. This will rather depend on the specific feature (attributes, CLI options, or combinations of both).